cmd/copy.go

```diff
@@ -148,15 +149,50 @@ func (raw rawCopyCmdArgs) blockSizeInBytes() uint32 {
 func (raw rawCopyCmdArgs) cook() (cookedCopyCmdArgs, error) {
 	cooked := cookedCopyCmdArgs{}
 
-	fromTo, err := validateFromTo(raw.src, raw.dst, raw.fromTo) // TODO: src/dst
+	var err error
+	cooked.fromTo, err = validateFromTo(raw.src, raw.dst, raw.fromTo) // TODO: src/dst
 	if err != nil {
 		return cooked, err
 	}
+
+	//Convert local paths to long filepaths on windows, get absolute filepath everywhere.
+	//This ensures that long file paths work on Windows and that relative filepaths using . work everywhere.
+	if cooked.fromTo.From() == common.ELocation.Local() {
+		raw.src, err = filepath.Abs(raw.src)
+		if err != nil {
+			return cooked, err
+		}
+
+		if common.OS_PATH_SEPARATOR == `\` {
+			raw.src = common.ToLongPath(raw.src)
+			common.AZCOPY_PATH_SEPARATOR_STRING, common.AZCOPY_PATH_SEPARATOR_CHAR = `\`, '\\'
+			if fi, err := os.Stat(raw.src); err == nil {
```

Comment:
I'm nervous about changing these.  I don't fully understand what Ze had in mind when he set these constants up; but I do seem to recall an assumption that they would in fact be constant and unchanging.  I.e. they didn't mean "what is the path separator that the current OS uses". They specifically meant "what is the path separator that AzCopy uses (regardless of the underyling OS)."

The difference may be significant because on the cloud side the separator is always / (i.e;. the old, hard-code value)

Reply:
Actually, Ze originally suggested shifting to this and seeing if there were any regressions. There were not.